### PR TITLE
Make bd_tools accessible through one interface

### PR DESCRIPTION
**Pull request by gbalke**
This enables installation of the `bd_tools` python package. Now all tools can be accessed via:
```
python3 -m bd_tools <tool_name>
```
The way I'm doing this is a bit hacky so if you're looking for help on `bd_tools` itself, the `-h` or `--help` must be the first arg.

### CODE REVIEW

File: bd_tools/__main__.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+BIN_PATH = Path(__file__).parent / "bin"
+
+
+def get_tools():
+    """Returns all scripts in the bin directory"""
+    tools = []
+    for tool in BIN_PATH.glob("*.py"):
+        tools.append(tool.stem)
+    return tools
```
Comment (brentyi) [inline bd_tools/__main__.py:14]: maybe more succinct as: `list(BIN_PATH.glob("*.py"))`

File: bd_tools/__main__.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def parser_args(tools):
+    parser = argparse.ArgumentParser(description="BetzDrive Tools Package")
+    parser.add_argument(
+        "tool", type=str, choices=tools, help=("Name of the tool to use.")
```
Comment (brentyi) [inline bd_tools/__main__.py:20]: any reason for the extra pair of parentheses in the help field?

File: bd_tools/__main__.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        code = compile(f.read(), file_name.name, "exec")
+        exec(code, globals())
+
+
+if __name__ == "__main__":
+    # NOTE(greg): We have to hack the help to make sure it only operates on
+    # this argparser if its the first arg.
+    tool_help = False
+    if "-h" in sys.argv or "--help" in sys.argv:
+        if sys.argv[1] == "-h" or sys.argv[1] == "--help":
```
Comment (brentyi) [inline bd_tools/__main__.py:42]: seems like this condition could be inverted

**Review by brentyi — APPROVED** [review]
This would make more sense to me as subparsers*, but if that's too involved only have a few minor comments!

*as a plug, we've also been working on an argparse + strong typing library that I think would make  the subparser pattern really easy/modular: http://github.com/brentyi/dcargs

File: bd_tools/__main__.py (hunk 1 above)
Comment (brentyi) [reply bd_tools/__main__.py:14]: oh, actually I just noticed the `.stem` sorry, maybe `[p.stem for p in BIN_PATH.glob("*.py")` or `list(map(lambda p: p.stem, BIN_PATH.glob("*.py")))`?

File: bd_tools/__main__.py (hunk 2 above)
Comment (gbalke) [reply bd_tools/__main__.py:20]: I had an append before that crossed to the newline. Can remove.

**Pull request merged**